File: src/com/temportalist/thaumicexpansion/common/ProxyCommon.java
package com.temportalist.thaumicexpansion.common;

import com.temportalist.thaumicexpansion.common.inventory.ContainerThaumicAnalyzer;
import com.temportalist.thaumicexpansion.common.tile.TEThaumicAnalyzer;
import cpw.mods.fml.common.network.IGuiHandler;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

/**
 * @author TheTemportalist
 */
public class ProxyCommon implements IGuiHandler {

	public void register() {
	}

	@Override
	public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x,
			int y, int z) {
		TileEntity tile = world.getTileEntity(x, y, z);
		if (tile != null) {
			if (ID == 0 && tile instanceof TEThaumicAnalyzer) {
				return new ContainerThaumicAnalyzer((TEThaumicAnalyzer) tile, player.inventory);
			}
		}
		return null;
	}

	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x,
			int y, int z) {
		return null;
	}

}

File: src/main/scala/com/temportalist/thaumicexpansion/common/ProxyCommon.scala
package com.temportalist.thaumicexpansion.common

import com.temportalist.origin.api.common.proxy.IProxy
import com.temportalist.thaumicexpansion.common.container.ContainerAnalyzer
import com.temportalist.thaumicexpansion.common.tile.TEAnalyzer
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World

/**
 *
 *
 * @author TheTemportalist
 */
class ProxyCommon extends IProxy {

	override def register(): Unit = {}

	override def getClientElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int,
			z: Int, tileEntity: TileEntity): AnyRef = null

	override def getServerElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int,
			z: Int, tileEntity: TileEntity): AnyRef = {
		if (tileEntity != null) tileEntity match {
			case te: TEAnalyzer =>
				return new ContainerAnalyzer(te, player)
			case _ =>
		}
		null
	}

}
